### DISCUSSION TITLE
Add custom override to Item Prices

### DISCUSSION THREAD

**Discussion by KogasaPls** [runelite/runelite#14903] [Ideas]
I suggest adding the option to manually override the price of items. Example use cases include:

* Setting a value of 0GP for items whose value is not meaningful, like untradeable items or items you will never sell
* Setting a value of currency (e.g. tokkul) based on how you intend to exchange them
* Setting a value of containers based on their expected value
* Reducing the value of artificially-inflated items (e.g. low-volume tradeable items)

To configure manual overrides, I'd suggest a text box containing lines of the form "{item} {price}", e.g. "vorkath's head 0". 

If you like the idea and think it's feasible without any headaches, I'd be happy to give it a try and submit a PR.

**Comment by Enriath** [comment]
* The value of a container could be set programatically, it's just quite a bit more complicated than the current 1:1 system.
* Artificially inflated items shouldn't stay excessively valuable for more than a short amount of time (like a few hours), and forcing users to mess with some config to mitigate that is pretty silly when it's somewhat possible to mitigate that.
* Tokkul and other variable currency is more of a problem due to how many options there are, and for anything that doesn't convert into something stable, setting by price directly would be insanely clunky for the user to deal with.
* Generally, a text box is a pretty clunky design, and requiring it to map item -> price makes the system inflexible when most situations will want to map item -> item somehow.

The big one here then is assigning arbitrary prices to stuff, which I still regard as silly. An item has value even if it's never going to be sold. The bank value isn't a value of stuff you're going to sell, it's closer to a "net worth" of your account. By allowing it to be an arbitrary price, it also opens up to people who just want to make themselves look cooler by inflating the value of their bank, which is wholly a net negative. While this can currently be done with low volume item manipulation, that's both public and temporary, so it isn't much of an issue.

Whether you like it or not, things have a value as defined by other players in a player run market. The bank already allows you to use multiple tabs, and we already offer bank tags and bank tag tabs which would effectively do this without lying to the user.

**Reply by KogasaPls** [reply]
>Artificially inflated items shouldn't stay excessively valuable for more than a short amount of time (like a few hours), and forcing users to mess with some config to mitigate that is pretty silly when it's somewhat possible to mitigate that.

My suggestion wouldn't force anyone to do anything. Users would have the same option, "don't use the override," that they have now.  If it's possible to mitigate it, then feel free to do so and cross this off the list of potential uses.

>Tokkul and other variable currency is more of a problem due to how many options there are, and for anything that doesn't convert into something stable, setting by price directly would be insanely clunky for the user to deal with.

It might be clunky, but this would offer some control where currently there is none. 

>Generally, a text box is a pretty clunky design, and requiring it to map item -> price makes the system inflexible when most situations will want to map item -> item somehow.

Mapping to some multiple of an item instead of a fixed price could of course be done. I don't know right away what a better configuration method would look like, except one that's much more complicated to implement.

>The big one here then is assigning arbitrary prices to stuff, which I still regard as silly. An item has value even if it's never going to be sold. 

An item has a grand exchange value even if it's never going to be sold. I'm not suggesting that your personal plans for an object affect the grand exchange value of the item. I'm suggesting that we allow users to use a different interpretation of "value" if they believe it is more useful to them. This is obviously personal and subjective, but that doesn't mean it's *worse* than the average exchange value. 

The average exchange value is easy to compute and easy to interpret as "usually a reasonable estimate of what you could exchange an item for." It is not *the best* estimate of what you could exchange any given item for (that would be impossible to determine), it's just the one that makes the most sense *in general, by default*. We've already seen examples where the current valuation clearly fails and could be improved on an individual basis (tokkul, Vorkath's head, item fluctuations, low volume items) even without straying from the interpretation of "value" as "a reasonable estimate of what you could exchange an item for."

>The bank value isn't a value of stuff you're going to sell, it's closer to a "net worth" of your account.

That is the only way to interpret the bank value as it is currently computed, but again, not the only reasonable way a bank value *could be* defined. I gave an example already, which is just "the expected exchange value of your bank if you exchange it in a particular way." This could mean replacing the value of ingredients/currency with the value of the corresponding outputs/products (which may or may not be 0). It could mean counting certain items' high alch value instead of their GE value. It could mean any number of things that are personally meaningful.

>By allowing it to be an arbitrary price, it also opens up to people who just want to make themselves look cooler by inflating the value of their bank, which is wholly a net negative.

Everyone can already easily lie about the value of their bank. You can call this a negative if you want, but it's barely a consequence at all. It's certainly not a *net* negative weighed against the added utility.

>Whether you like it or not, things have a value as defined by other players in a player run market. 

Do you think I believe that the current valuation doesn't exist? Not to mention, this completely ignores the examples of untradeable items with a value and tradeable items with no value.

>The bank already allows you to use multiple tabs, and we already offer bank tags and bank tag tabs which would effectively do this without lying to the user.

Bank tabs and tags let you easily compute the value of some items but not others. That is not really close to the scope of what would be possible with this suggestion.

**Comment by gen-ux** [comment]
+1 for this idea. Here's more context and justification for why I think this would be a valuable addition.

I play ironman mode only and while GE value is a decent approximation of item value to an ironman, it has some issues. For example, I use the default 7k threshold for the first tier of highlighted items, and that does a good job 90% of the time. However, a grimy ranarr weed is often times just above or just below this value. I don't care what the value is at any particular day - a ranarr is worth one ranarr and to me, it 's valuable enough for that >7k category. That means I either adjust the threshold and mess with the categories for lots of other items or I use the highlight item feature, but that feature doesn't account for quantity. I would love to just be able to set {grimy ranarr weed}{8000} in some key:value pair list and be done with it. I would do the same for other herbs, seeds, and many other items.

Another situation where I would love this feature is items with a stable GE value that's massively higher than its value to an ironman. Take dragonfruit tree seeds as an example: I'm 99 farming so I don't give a crap about these things. I would happily trade a dragonfruit tree seed for a snapdragon seed, even though a dragonfruit tree seed is >4x the GE price of a snapdragon seed. With this feature, I could manually set a much lower value for items like dragonfruit tree seeds.

And here's one more case that the OP touched on: untradeable items with poorly defined value. Examples include clue scrolls, frozen cache, dark totem pieces, ancient shards, etc. Right now I just highlight all of these, but an elite clue scroll clearly has more value than a beginner clue scroll but the highlight feature makes them all appear equal. It would be nice to be able to categorize these more accurately.

**Comment by MosheBenZacharia** [comment]
if this ever gets revisited I have an implementation of remapping some untradeable items to their tradeable counterparts for reference: https://github.com/MosheBenZacharia/inventorytotal/blob/master/src/main/java/com/ericversteeg/ValueRemapper.java

**Comment by TV-cmd** [comment]
Best idea. Why is graceful counted in the bank value when even if I were to liquidify my bank I would never "sell" my graceful pieces to get amalyse
